packages/manager/src/managers/slices/SlicesManager.ts

```diff
@@ -641,6 +641,9 @@ export class SlicesManager extends BaseManager {
 	): Promise<SharedSlice> {
 		const sliceMachineConfig = await this.project.getSliceMachineConfig();
 
+		// Update the AWS ACL before uploading screenshots as it might have expired
+		await this.screenshots.initS3ACL();
+
```

Comment:
I'd maybe move that down the stream right to the `uploadScreenshot` method, this way if there's no new screenshot we don't even need to bother with initing the ACL and other methods relying on `uploadScreenshot` would not fail too.

Reply:
Sounds good to me, just a heads up though that this *might* not because this method is called in parallel inside `pushChanges`: https://github.com/prismicio/slice-machine/blob/bmo/fix-init-s3-acl/packages/manager/src/managers/prismicRepository/PrismicRepositoryManager.ts#L262

If you want to be sure the ACL is inited only once, you could instead move it there, before the line I linked to.

Reply:
You're right @xrutayisire, that's a compromise